src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CMDLineProgramModule.java

```diff
@@ -0,0 +1,124 @@
+
+        if(0!=exitStatus){ onError(commands, stderrMessageDestination, exitStatus); }
+    }
+
+    protected void setupWorkingDir(final ProcessBuilder builder,
+                                   final File directoryToWorkIn,
+                                   final File stdoutDestination){
+        builder.directory(directoryToWorkIn);
+        if(null!=stdoutDestination) { builder.redirectOutput(stdoutDestination); }
+    }
+
+    /**
+     * Sets up working environment for the program. Inheriting classes should provide implementation.
+     * @param builder process builder for the program
+     * @param args    arguments used for setting up the environment
+     */
+    abstract void setupWorkingEnvironment(final ProcessBuilder builder, final String... args);
+
+    /**
+     * Redirects the stderr message from the represented program to a string.
+     * @param process       the program's process
+     * @return              String converted from the stderr message of the program
+     * @throws IOException
+     */
+    protected static String redirectSTDERR(final Process process) throws IOException{
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+        StringBuilder out = new StringBuilder();
+        String line       = null;
+        String previous   = null;
+        while ((line = reader.readLine()) != null) {
+            if (!line.equals(previous)) {
+                previous = line;
+                out.append(line).append('\n');
+            }
+        }
+        return out.toString();
+    }
+
+    /**
+     * Strategy for error handling when program returns non-zero exit status.
+     * @param commands           the list of commands, arguments, flags used in running the process
+     * @param stderrMessage      the stderr message form the process
+     * @param commandExitStatus  exit status of the process
+     * @throws RuntimeException
+     */
+    private static void onError(final ArrayList<String> commands, final String stderrMessage, final int commandExitStatus) throws RuntimeException{
+        String errorMessage = "";
+        for(final String mess : commands){ errorMessage += " " + mess; }
+        errorMessage += "\n" + stderrMessage;
+        throw new RuntimeException("Errors occurred while running: " + errorMessage +
+                                    "\nWith exit status: " + commandExitStatus);
+    }
+
+    /**
+     * Checks if the program to be represented is available.
+     * @param programName   the name of the program (case-sensitive) to be tested and run subsequently if available
+     * @throws IOException  throws IOException if the program is not available.
+     */
+    @VisibleForTesting
+    static void checkIfProgramIsAvailableOnHost(final String programName) throws IOException{
+        try{
+            final ArrayList<String> commands = new ArrayList<>();
+            commands.add("which");
```

Comment:
Actually, I think this class should not depend on the `$PATH` variable but require a fully qualified path to the executable. It's much cleaner that way and eliminates one source of potential errors when someone has misconfigured `$PATH`.